### PR TITLE
Updated MasterTestPlan.md and added subordinate files

### PR DESCRIPTION
**Pull request by matsduf**
* Copied Algorithms-used-by-dnscheck.md from DNSCheck Wiki to Zonemaster repository
* Copied Tests.md from DNSCheck Wiki to Zonemaster repository (new name is DNSCheck-Tests.md)
* Corrected link in MasterTestPlan.md
* Updated links in MasterTestPlan.md from pointing at files in DNSCheck Wiki to the local files 
* Minior editorial changes of MasterTestPlan.md
* Corrected markdown encoding of headlines
* Updated links to right format
* Corrected second level i lists

### CODE REVIEW

**Comment by matsduf** [comment]
@sandoche2k, can you review this PR?

**Comment by mattias-p** [comment]
@sandoche2k I think there is a good argument why the documents should not be internal. The master test plan links to the documents, and the master test plan is not internal, so the documents shouldn't be either.

**Comment by sandoche2k** [comment]
Maybe in an another directory. But not in the requirements directory. We
could provide a link as it is now from the Master Test Plan.

**Comment by mattias-p** [comment]
They're mentioned in the master test plan under system requirements and as external references. What do you think of having them under docs/requirements but clarifying their status in the README?

**Comment by matsduf** [comment]
@sandoche2k, the reason that I put the documents in "docs/requirement" instead of "docs/internal-documentation/requirements" are two fold:
* The documents are linked from external documents.
* The primary linking is from documents in the "requirements" directory.

I will make a new suggestion.

**Comment by matsduf** [comment]
@sandoche2k, please review my updates.

File: docs/requirements/TestRequirements.md [hunk 1]
```diff
@@ -130,13 +130,13 @@ Requirements on writing test specifications
 These are some requirements for writing specifications for "Zonemaster":
 
  1. Follow the framework of the IEEE 829-2008.
- 2. The documents must be in [Markdown Syntax]
-    (http://daringfireball.net/projects/markdown/syntax).
+ 2. The documents must be in 
+    [Markdown Syntax](http://daringfireball.net/projects/markdown/syntax).
```
Comment (sandoche2k) [inline docs/requirements/TestRequirements.md]: Why  [Markdown Syntax] in newline ?

Comment (matsduf) [reply docs/requirements/TestRequirements.md]: There must not be any white space or line break between "]" and "(". There are three options that give the same result on Github:

```
 2. The documents must be in 
     [Markdown Syntax](http://daringfireball.net/projects/markdown/syntax).

 2. The documents must be in [Markdown Syntax
     ](http://daringfireball.net/projects/markdown/syntax)

 2. The documents must be in [Markdown Syntax](
     http://daringfireball.net/projects/markdown/syntax)
```

Comment (mattias-p) [reply docs/requirements/TestRequirements.md]: There's also:

```
The documents must be in [Markdown Syntax].

[Markdown Syntax]: http://daringfireball.net/projects/markdown/syntax
```

https://spec.commonmark.org/0.28/#reference-link

**Pull request merged**